internal/wrphandlers/mocktr181/handler.go

```diff
@@ -0,0 +1,211 @@
+// SPDX-FileCopyrightText: 2024 Comcast Cable Communications Management, LLC
+// SPDX-License-Identifier: Apache-2.0
+
+package mocktr181
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/xmidt-org/wrp-go/v3"
+	"github.com/xmidt-org/xmidt-agent/internal/wrpkit"
+	"go.uber.org/zap"
+)
+
+var (
+	ErrInvalidInput     = fmt.Errorf("invalid input")
+	ErrInvalidFileInput = fmt.Errorf("misconfigured file input")
+	ErrUnableToReadFile = fmt.Errorf("unable to read file")
+)
+
+// Option is a functional option type for WS.
+type Option interface {
+	apply(*Handler) error
+}
+
+type optionFunc func(*Handler) error
+
+func (f optionFunc) apply(c *Handler) error {
+	return f(c)
+}
+
+// Handler sends a response when a message is required to have a response, but
+// was not handled by the next handler in the chain.
+type Handler struct {
+	next       wrpkit.Handler
+	egress     wrpkit.Handler
+	source     string
+	filePath   string
+	parameters []MockParameter
+	logger     *zap.Logger
+}
+
+type MockParameter struct {
+	Name       string
+	Value      string
+	Access     string
+	DataType   int // add json labels here
+	Attributes map[string]interface{}
+	Delay      int
+}
+
+type MockParameters struct {
+	Parameters []MockParameter
+}
+
+type Tr181Payload struct {
+	Command    string      `json:"command"`
+	Names      []string    `json:"names"`
+	Parameters []Parameter `json:"parameters"`
+}
+
+type Parameters struct {
+	Parameters []Parameter
+}
+
+type Parameter struct {
+	Name       string                 `json:"name"`
+	Value      string                 `json:"value"`
+	DataType   int                    `json:"dataType"`
+	Attributes map[string]interface{} `json:"attributes"`
+}
+
+// New creates a new instance of the Handler struct.  The parameter next is the
+// handler that will be called and monitored for errors.  The parameter egress is
+// the handler that will be called to send the response if/when the next handler
+// fails to handle the message.  The parameter source is the source to use in
+// the response message.
+func New(next, egress wrpkit.Handler, source string, logger *zap.Logger, opts ...Option) (*Handler, error) {
+	// TODO - load config from file system
+
+	h := Handler{
+		next:   next,
+		egress: egress,
+		source: source,
```

Comment:
looks like you forgot to add the logger

I think using `logger := logger.Named("mocktr181")` would be helpful during debugging

Reply:
i think adding the logger as an option is better

Reply:
oops, thanks, good catch